elixir/apps/domain/test/domain/auth/adapters/openid_connect_test.exs

```diff
@@ -304,57 +304,6 @@ defmodule Domain.Auth.Adapters.OpenIDConnectTest do
       assert DateTime.diff(identity.provider_state["expires_at"], DateTime.utc_now()) in 3595..3605
     end
 
-    test "prefers existing identities over manually created identities on a conflict", %{
```

Comment:
@jamilbk, double check me here on this deletion.  I'm fairly certain there is no way we'd get in to the situation now that we've got an explicit `email` column, but want to see if you could think of a possible scenario I am not thinking of right now.

Reply:
These could even be different actors.

I think the likelihood of this occurring is super low, but might want to have a test for this just we know what the expected behavior is (it might even be "we fail with 500")

Reply:
In the scenario you laid out, we'd both not upsert the email and also break the sign in, because the sign in is dependent on the upsert to succeed.

I agree we probably should have a test knowing the outcome of this scenario.  At the same time, this scenario means the customer would be reusing email addresses in their IdP system, which seems like it could cause all kinds of problems beyond just Firezone.

In an ideal world, what would your preferred outcome be for this situation?

Reply:
Just want to make sure we have the edge case documented somewhere in case we pull in a large customer and they hit this. We seem to have a knack for hitting IdP edge cases.

Reply:
Yeah, totally understandable.  I'll see if we can avoid the 500 and at least return a message saying "please contact your admin" or something along those lines.  Worst case, like you said, we 500 for now, but we've at least got a test for it.

Reply:
Ok, I believe I've addressed the edge case we've discussed here.  In the rare event this happens, the user(s) will get a message on sign in saying `You may not authenticate to this account.`

It's not perfect, but it doesn't throw a 500 error and it doesn't allow unintended logins if an email has been reused on the IdP side.